mteb/evaluation/evaluators/utils.py

```diff
@@ -266,9 +293,85 @@ def evaluate_change(
     changes_df = pd.DataFrame(changes)
     changes_df["p-MRR"] = changes_df.apply(lambda x: rank_score(x), axis=1)
     qid_wise = changes_df.groupby("qid").agg({"p-MRR": "mean"})
-    return {
-        "p-MRR": qid_wise["p-MRR"].mean(),
+    return qid_wise["p-MRR"].mean()
+
+
+def evaluate_p_mrr_change(
+    results: dict[str, dict[str, float]],
+    qrels: dict[str, dict[str, float]],
+    task_name: str,
+    k_values: list[int],
+) -> dict[str, float]:
+    """Computes the scores needed for FollowIR datasets, including p-MRR (measuring change in instruction) and
+    details about the original instruction run and changed instruction run.
+    """
+    followir_scores = defaultdict(dict)
+    # load the qrel_diff from the dataset
+    TASK_TO_HF_DATASET = {
+        "Core17InstructionRetrieval": ("jhu-clsp/core17-instructions-mteb", False),
+        "Robust04InstructionRetrieval": ("jhu-clsp/robust04-instructions-mteb", False),
+        "News21InstructionRetrieval": ("jhu-clsp/news21-instructions-mteb", False),
+        "mFollowIR": ("jhu-clsp/mfollowir-parquet-mteb", True),
+        "mFollowIRCrossLingual": (
+            "jhu-clsp/mfollowir-cross-lingual-parquet-mteb",
+            True,
+        ),
     }
+    hf_path, is_multilingual = TASK_TO_HF_DATASET[task_name]
+    if is_multilingual:
+        # figure out which of the languages this is: ["zho", "rus", "fas"]
+        # gather the changed_qrels for each, and store the keys as a check
+        for lang in ["zho", "rus", "fas"]:
```

Comment:
Can this be loaded in task class?

Reply:
It could be but then I'd have to define a custom loader. I was leaning towards not, since this is just the info needed purely for eval and not for anything else. If you have a strong preference for putting it in the task class I can move it, but I figured the current setup leaves the `tasks` section of the code cleaner.